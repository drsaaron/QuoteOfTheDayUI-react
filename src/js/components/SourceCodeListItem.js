/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import React, {Component} from 'react';
import { connect } from 'react-redux';
import { retrieveQuotesForSourceCode } from '../actions/SourceCodeActions';

class SourceCodeListItem extends Component {
    constructor(props) {
        super(props);
        
        this.clickHandler = this.clickHandler.bind(this);
    }
    
    clickHandler(event) {
        event.preventDefault();
        this.props.retrieveQuotesForSourceCode(this.props.sourceCode);
    }

    render() {
        return (
                <li><a href="#" onClick={this.clickHandler}>{this.props.sourceCode.text}</a></li>
                );
    }
}

const mapStateToProps = (state) => {
    return {
        
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        retrieveQuotesForSourceCode: (sourceCode) => dispatch(retrieveQuotesForSourceCode(sourceCode))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SourceCodeListItem);