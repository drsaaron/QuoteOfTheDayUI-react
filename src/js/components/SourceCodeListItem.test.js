import React from 'react';
import renderer from 'react-test-renderer';

import SourceCodeListItem from './SourceCodeListItem';

describe('SourceCodeListItem', () => {
    test('basic', () => {
	var sourceCode = { text: "I am some text" };
	const component = renderer.create(<SourceCodeListItem sourceCode={sourceCode} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
