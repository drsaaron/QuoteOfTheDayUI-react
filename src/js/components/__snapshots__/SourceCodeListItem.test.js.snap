// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceCodeListItem basic 1`] = `
<li>
  <a
    href="/"
    onClick={[Function]}
  >
    I am some text
  </a>
   
  <button
    onClick={[Function]}
  >
    +
  </button>
</li>
`;
